### PR TITLE
RSDK-8124 Cleanup entry point naming

### PR DESCRIPTION
**Pull request by acmorrow**
Should be no behavior changes, just naming and bringing mainlines into closer agreement ahead of simplifying them.

- `serve_web_with_external_network` moved to common as `serve_with_network`
- `serve_web` to `serve`
- Reorder `native/main.rs` to be more similar to `esp32/main.rs`
- `serve_web_inner` to `serve_inner`
- `serve_async_with_external_network` to `async_serve_with_network`
- `serve_async` to `async_serve`
- Reorder project template `main.rs` to be more similar to `esp32/main.rs`

The next push will be to make `async_serve` use `async_serve_with_network`, and then fold `serve_inner` into `async_serve_with_network`.

### CODE REVIEW

File: micro-rdk-server/native/main.rs [hunk 1]
```diff
@@ -3,46 +3,55 @@ mod native {
...
-                    ROBOT_SECRET.unwrap().to_string(),
-                )
-                .into(),
-            ) {
-                log::error!("Failed to store RobotCredentials: {}", e);
+
+        // At runtime, if the program does not detect credentials or configs in storage,
+        // it will try to load statically compiled values.
+
+        if !storage.has_robot_configuration() {
```
Comment (acmorrow) [inline micro-rdk-server/native/main.rs:32]: This change is only to make this block identical to the one in esp32/main.rs.

File: micro-rdk-ffi/src/ffi/runtime.rs [hunk 2]
```diff
@@ -271,38 +273,23 @@ pub unsafe extern "C" fn viam_server_start(ctx: *mut viam_server_context) -> via
...
-            let storage = micro_rdk::esp32::nvs_storage::NVSStorage::new("nvs").unwrap();
-            micro_rdk::esp32::entry::serve_web_with_external_network(
-                Some(ctx.provisioning_info),
-                repr,
-                max_connection,
-                storage,
-                network,
-            )
-        }
+        let storage = micro_rdk::esp32::nvs_storage::NVSStorage::new("nvs").unwrap();
```
Comment (mattjperez) [inline micro-rdk-ffi/src/ffi/runtime.rs:284]: nice :sunglasses:, didn't see the only difference was just the value of the storage passed in

File: micro-rdk/src/native/entry.rs [hunk 3]
```diff
@@ -1,41 +1,3 @@
...
-        exec,
-        info,
-        storage,
-        repr,
-        network,
-        max_webrtc_connection,
-    )));
-}
-
 #[cfg(test)]
```
Comment (mattjperez) [inline micro-rdk/src/native/entry.rs:1]: not sure if we should rename or move the code into another file at this point

**Review by mattjperez — APPROVED** [review]
LGTM!

**Review by npmenard — APPROVED** [review]
LGTM

**Pull request merged**